src/test/resources/features/actions/line_mover.feature

```diff
@@ -62,36 +89,213 @@ Feature: Line mover
       | checkpoint  | input          | ""      | # comment |
       | subworkflow | workdir        | ""      | # comment |
 
-  Scenario Outline: Permutation down between rule and comment sections
+  Scenario Outline: Permutation up/down inside rule section
+    Given a snakemake project
+    Given I open a file "foo1.smk" with text
+    """
+    <rule_like1> NAME1:
+        <rule_section1>: <content1>
+        <rule_section2>: <content2>
+    """
+    When I put the caret at <rule_section2>: <content2>
+    And I invoke MoveStatementUp action
+    Then editor content will be
+    """
+    <rule_like1> NAME1:
+        <rule_section2>: <content2>
+        <rule_section1>: <content1>
+
+    """
+    When I put the caret at <rule_section2>: <content2>
+    And I invoke MoveStatementDown action
+    Then editor content will be
+    """
+    <rule_like1> NAME1:
+        <rule_section1>: <content1>
+        <rule_section2>: <content2>
+
+    """
+    Examples:
+      | rule_like1  | rule_section1  | rule_section2 | content1        | content2         |
+      | rule        | input          | output        | "file.txt"      | "file2.txt"      |
+      | checkpoint  | input          | output        | "file.txt"      | "file2.txt"      |
+      | subworkflow | workdir        | configfile    | "/"             | "/"              |
+      | rule        | input          | output        | "file.txt"      | "file2.txt"      |
+      | checkpoint  | input          | output        | "file.txt"      | "file2.txt"      |
+      | subworkflow | workdir        | configfile    | "/"             | "/"              |
+      | rule        | input          | output        | "file.txt"      | "file2.txt"      |
+      | checkpoint  | input          | output        | "file.txt"      | "file2.txt"      |
+      | subworkflow | workdir        | configfile    | "/"             | "/"              |
+
+  Scenario Outline: Permutation up/down inside rule section with complex content
     Given a snakemake project
     Given I open a file "foo1.smk" with text
     """
     <rule_like1> NAME1:
         <rule_section1>:
```

Comment:
This rule could be merged with the previous one if you use `\n` in content, e.g. you could use values like input: "foo" or input:\n "foo" to make section with content single/multi line